### ISSUE TITLE
document_etag fails in delete

### ISSUE THREAD

**Issue by arieb** [RedTurtle/eve-sqlalchemy#9]
I Have a simple User model that contains a relationship with a Role model.

```python
roles_users = db.Table('roles_users',
                       db.Column('user_id', db.Integer, db.ForeignKey('user.id')),
                       db.Column('role_id', db.Integer, db.ForeignKey('role.id')))

class Role(EveModel):
    id = db.Column(db.Integer(), primary_key=True)
    name = db.Column(db.String(80), unique=True)
    description = db.Column(db.String(255))

class BaseIdentity(EveModel, UserMixin):
    __tablename__ = 'base_identity'
    id = db.Column(db.Integer, primary_key=True)
    id_type = db.Column(db.String(32), nullable=False)
    username = db.Column(db.String(255), unique=True, nullable=False)
    password = db.Column(db.String(255), nullable=False)

    __mapper_args__ = {
        'polymorphic_identity': 'base_identity',
        'polymorphic_on': id_type
    }
class User(BaseIdentity):
    __tablename__ = 'user'
    id = db.Column(db.Integer, db.ForeignKey('base_identity.id'), primary_key=True, nullable=False)
    firstname = db.Column(db.String(255), nullable=False)
    lastname = db.Column(db.String(255), nullable=False)
    email = db.Column(db.String(255), nullable=False)

    roles = db.relationship('Role',
                            secondary=roles_users,
                            backref=db.backref('users', lazy='dynamic'))
    __mapper_args__ = {
        'polymorphic_identity': 'user',
    }
```

EveModel implements the jsonify method from the examples.
HATEOS is disabled.

when i try to delete a user using:

curl -v -H "If-Match: 771a2d531414a0819b45f7432b1e1334701e4ad5" -X DELETE --user admin http://localhost:5000/api/v1/user/3  

i get the following exception:

```python
.... cropped for brevity ......
 File "/home/user/git/simple_site/venv/lib/python2.7/site-packages/Eve-0.5-py2.7.egg/eve/methods/common.py", line 794, in decorated
    r = f(resource, **combined_args)
  File "/home/user/git/simple_site/venv/lib/python2.7/site-packages/Eve-0.5-py2.7.egg/eve/methods/delete.py", line 33, in deleteitem
    return deleteitem_internal(resource, concurrency_check=True, **lookup)
  File "/home/user/git/simple_site/venv/lib/python2.7/site-packages/Eve-0.5-py2.7.egg/eve/methods/delete.py", line 76, in deleteitem_internal
    original = get_document(resource, concurrency_check, **lookup)
  File "/home/user/git/simple_site/venv/lib/python2.7/site-packages/Eve-0.5-py2.7.egg/eve/methods/common.py", line 64, in get_document
    etag = document.get(config.ETAG, document_etag(document))
  File "/home/user/git/simple_site/venv/lib/python2.7/site-packages/Eve-0.5-py2.7.egg/eve/utils.py", line 277, in document_etag
    h.update(dumps(value, sort_keys=True).encode('utf-8'))
  File "build/bdist.linux-x86_64/egg/bson/json_util.py", line 125, in dumps
    return json.dumps(_json_convert(obj), *args, **kwargs)
  File "/usr/lib/python2.7/json/__init__.py", line 250, in dumps
    sort_keys=sort_keys, **kw).encode(obj)
  File "/usr/lib/python2.7/json/encoder.py", line 209, in encode
    chunks = list(chunks)
  File "/usr/lib/python2.7/json/encoder.py", line 434, in _iterencode
    for chunk in _iterencode_dict(o, _current_indent_level):
  File "/usr/lib/python2.7/json/encoder.py", line 408, in _iterencode_dict
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 332, in _iterencode_list
    for chunk in chunks:
  File "/usr/lib/python2.7/json/encoder.py", line 442, in _iterencode
    o = _default(o)
  File "/usr/lib/python2.7/json/encoder.py", line 184, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: <simple_site.models.user.Role object at 0x7fc1146c9e90> is not JSON serializable
```

shouldn't the model be parsed by the DataLayer's parser?

**Comment by arieb** [comment]
@amleczko any update on this? should this be resolve here or in the eve repo?

**Comment by Leonidaz0r** [comment]
Is this bug still relevant? It might have been solved in edf8c63280393c7aaaf28530a1a725f12fe8ba2d

**Comment by amleczko** [comment]
You are right, I have tested it with current master and it's fixed.
